cosmicfishpie/cosmology/nuisance.py

```diff
@@ -179,38 +305,28 @@ def gcsp_bias(self):
         --------
         Reads from file and interpolates for a given survey
         """
-        # this dict can be read from a file
-        # InterpolatedUnivariateSpline allows for extrapolation outside
-        # bounds of the input files. Uses order=1 linear splines.
-        bofz_spec = InterpolatedUnivariateSpline(self.gc_table[:, 1], self.gc_table[:, 4], k=1)
+        bias_at_zmids = self.gcsp_bias_at_zm()
+        z_mids = self.gcsp_zbins_mids()
+        bofz_spec = InterpolatedUnivariateSpline(z_mids, bias_at_zmids, k=1)
         return bofz_spec
 
-    def gcsp_dndz(self):
-        """
-        Reads from file for a given survey
-        """
-        # this dict can be read from a file
-        dndz = self.gc_table[:, 3]
-        return dndz
+    def gcsp_sigmapv_at_z(self, z, sigma_key="sigmap"):
```

Comment:
These should be renamed to something like rescaling factor or simmilar. Because it actually does not return what the name says